index.php

```diff
@@ -45,7 +45,10 @@
 
 	foreach (scandir("system/core/") as $_file) if (substr($_file, -4) == '.php') require_once("system/core/" . $_file);
 
+	$pageTypeIndex = count(get_declared_classes()) -1;
 	foreach (scandir("system/datatypes/") as $_file) if (substr($_file, -4) == '.php') include_once("system/datatypes/" . $_file);
+	$hyphaPageTypes = [];
+	foreach (array_slice(get_declared_classes(), $pageTypeIndex) as $class) if (in_array('Page', class_parents($class)) && (new \ReflectionClass($class))->isInstantiable()) $hyphaPageTypes[] = $class;
```

Comment:
Interesting approach to autodetect datatype classes. Not sure if I really find it elegant or hacky, though :-p

In any case, this change would be better in a separate commit, or should at least be mentioned by the commit message.

Reply:
I understand that it looks a bit hacky but I do believe this to be less error prone.

I've put it in a separate commit.